request/request.js

```diff
@@ -155,6 +155,13 @@ module.exports = function($window, Promise, oncompletion) {
 				}
 			}
 
+			xhr.ontimeout = function (ev) {
+				if (aborted) return;
+				var error = new Error("Request timeout");
+				error.timeout = ev.target.timeout;
+				reject(error);
+			}
```

Comment:
Have you verified the `onreadystatechange` callback isn't also executed? If it is, you'll need to set `aborted` to `true` to keep anything else from firing.